Refactor Plugin to pass project/sort/limit/etc to bson-transpilers (#27)

aherlihy opened the pull request:
<!-- Ticket number and a general summary of your changes in the Title above -->
<!-- e.g. COMPASS-1111: updates ace editor width in agg pipeline view -->

<!--- The following fields are not obligatory. Use your best judgement on what you think is applicable to the work you've done -->

## Description
<!--- Describe your changes in detail -->
Pass the string for the URI, filter, projection, sort, skip, limit, maxTimeMS, collation instead of just the filter/agg.
<!--- If applicable, describe (or illustrate) architecture flow -->
### Checklist
- [x] New tests and/or benchmarks are included
- [x] Documentation is changed or added

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
Part of a "quick win" to export some additional drivers syntax.

<!--- If it's updating a dependancy, link to the Pull Request that originally introduced the fix -->
- [ ] Bugfix
- [x] New feature
- [ ] Dependency update
- [ ] Misc

## Open Questions
<!--- Any particular areas you'd like reviewers to pay attention to? -->

## Dependents
<!--- If applicable, link PRs/commits that this PR is dependent on or is a dependency of. -->
https://github.com/mongodb-js/compass-query-bar/pull/23
https://github.com/mongodb-js/bson-transpilers/pull/134

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] *Backport Needed*
- [ ] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [x] Major (fix or feature that would cause existing functionality to change)

aherlihy commented on electron/renderer/components/export-to-language-standalone.jsx:
```diff
@@ -10,7 +10,19 @@ class ExportToLanguageStandalone extends Component {
+          {filter: event.target.value}
+        );
+      } catch(err) {
+        console.log('Invalid input:' + err.message);
```
This is just the skeleton component that opens up the actual components. I think its better to catch the error and log it to the console than more work to make it render the error "prettily", since it's only ever used to test it locally

lrlna replied:
ah gotcha!

aherlihy commented:
That is how you declare variables in java (and C, C++, etc), in most statically typed languages you declare variables by writing \<type\> \<variable name\> = \<content\>. If you're curious: https://www.w3schools.com/java/java_variables.asp

When you say lack of a break between line 3/4, you mean a lack of break between the `new MongoClient` and the `new MongoClientURI`?

So you'd want it to look like:

```
const connection = MongoClient mongoClient = new MongoClient(
    new MongoClientURI(
            <uri>
    )
);
```
Right? It's super easy to change, just a driver template change :)

lrlna commented:
@aherlihy didn't know that was common variable declarations. I guess my only typed language is Rust, and it does `let variable_name: Type = "blah blah";`! so i was confused where `Bson` situation was coming from.

The line break I was more thinking like:
```js
const filter = {
  'x': 'y'
};

MongoClient.connect()
```
You have line breaks with both python and java filter declarations, but not with js (i think).

aherlihy commented:
I understand! I've updated the template in bson-transpilers, rereleased and updated here so that there's now a newline between variable declarations and connection in javascript.

And yeah, surprised that rust went with something different but I'm sure they have a good reason!
